### ISSUE TITLE
Expose the resolve function to other plugins

### ISSUE THREAD

**Issue by arcanis** [benmosher/eslint-plugin-import#233]
Hi,

I'm writing an eslint plugin that should check a few things with the modules, and I feel like it would be interesting to use the eslint-plugin-import resolver instead of rolling my own (especially since I use eslint-plugin-import too). I could copy/paste your code, but I feel like it would be much better to simply use the "standard implementation" right at the source (it would allow me to have a perfect compatibility with your module as far as the configuration goes). It would be as simple as this:

```js
var resolve = require('eslint-plugin-import').resolve;

module.exports = function (context) {
    resolve('some-module', context);
};
```

Would you mind consider exposing the `resolve` function to the world?

(Worth noting that I have seen that the resolvers are available as standalone packages - my point is mainly about exposing the eslint configuration parser rather than the resolvers themselves)

**Comment by benmosher** [comment]
Ah, neat! Yeah, I would be up for packaging it separately. I'll take a look tomorrow. 

I can imagine basically breaking the whole `core` folder out into a package.

Is Node 0.x support important to you? Or is Node 4+ okay?

**Comment by arcanis** [comment]
Node 4+ is fine ! :smile:

**Comment by benmosher** [comment]
Published v0.1.0 of `eslint-module-utils` to npm this morning. Can't guarantee the signatures and I haven't migrated the tests yet, but the fundamental functionality should remain fairly constant. 

I'm going to try to prove it out and document everything over the next week or two. Let me know how it goes leveraging it for what you're doing. 

I think I am going to pull some of the traversals out as well, like the stuff from no-unresolved that finds ES, CommonJS, and AMD module references and maybe the stuff that walks namespace MemberExpressions. So BOLO. 😄

**Comment by benmosher** [comment]
FYI: planning to roll back some things and support Node 0.10 and 0.12 until ESLint doesn't.

**Comment by arcanis** [comment]
Thanks for the heads up !

**Comment by benmosher** [comment]
Well, never got around to making it support 0.10 and 0.12, so moving forward with it as-is. Watch #408 for progress!
